flang/lib/Frontend/FrontendActions.cpp

```diff
@@ -582,8 +643,12 @@ void BackendAction::ExecuteAction() {
       ? llvm::CodeGenFileType::CGFT_AssemblyFile
       : llvm::CodeGenFileType::CGFT_ObjectFile;
   if (TM->addPassesToEmitFile(CodeGenPasses,
-          ci.IsOutputStreamNull() ? *os : ci.GetOutputStream(), nullptr, cgft))
-    assert(false && "Something went wrong");
+          ci.IsOutputStreamNull() ? *os : ci.GetOutputStream(), nullptr, cgft)) {
+    unsigned diagID = ci.diagnostics().getCustomDiagID(
+        clang::DiagnosticsEngine::Error, "emission of this file type is not supported");
+    ci.diagnostics().Report(diagID);
+    return;
```

Comment:
See question above.

Reply:
Will delete. There's another one that I will also delete.